feat: Create upgrading tool (#988)
vaadin/docs: articles/upgrading/flow/17-18.asciidoc

[discrete]
==== Flow Breaking Changes

- `AppShellRegistry` method `getTitle()` is removed 
It was broken and could not work. Instead, if needed, use `getUI().getUIInternals().getAppShellTitle()`. 

> - Having the `@Theme` annotation on Flow views or router layouts will not be allowed anymore, it should be on `AppShellConfigurator` instead. This is now consistent with the `@PWA` annotation and cleaner anyway since you can only have one `@Theme` per application anyway. 

Reviewer: ```suggestion
- Having the `@Theme` annotation on Flow views or router layouts will no longer be allowed.
The annotation should be on [interfacename]#AppShellConfigurator# instead. 
This is now consistent with the `@PWA` annotation.
It is also cleaner, since you can only have one `@Theme` per application. 

```
